api/src/main/java/org/openmrs/api/impl/OrderServiceImpl.java

```diff
@@ -292,6 +293,28 @@ private void ensureCareSettingIsSet(Order order, OrderContext orderContext) {
 		order.setCareSetting(careSetting);
 	}
 
+	private void validateFieldsForOrderContextOutPatientCareSettingType(Order order, OrderContext orderContext) {
```

Comment:
I don't really like that this duplicates validation logic already happening elsewhere (and would need to be kept consistent with it), or that this is limited to this very specific issue that was discovered in the ticket.

My suggestion in the ticket was to try invoking the validator in the saveOrderInternal method, before saving with the DAO.  Would be good to know if this was attempted or not and if it caused problems, or otherwise why that approach was not taken?